inc/class-wpseo-content-images.php

```diff
@@ -70,10 +52,9 @@ private function get_images_from_content( $content ) {
 		foreach ( $this->get_img_tags_from_content( $content ) as $img ) {
 			$url = $this->get_img_tag_source( $img );
 			if ( $url ) {
-				$images[ $url ] = WPSEO_Image_Utils::get_attachment_by_url( $url );
+				$images[ $url ] = $url;
```

Comment:
This feels like a nasty way to filter out duplicates. I'd much rater see an early return if no URL is present and add an extra `if` to ensure that no duplicates can be added.